crates/file_store/src/store.rs

```diff
@@ -63,17 +48,24 @@ where
         })
     }
 
-    /// Open an existing [`Store`].
+    /// Load an existing [`Store`].
     ///
-    /// Use [`create_new`] to create a new `Store`.
+    /// Use [`create`] to create a new [`Store`].
     ///
     /// # Errors
     ///
-    /// If the prefixed bytes of the opened file does not match the provided `magic`, the
-    /// [`FileError::InvalidMagicBytes`] error variant will be returned.
+    /// If the prefixed bytes of the loaded file does not match the provided `magic`, a
+    /// [`StoreErrorWithDump`] will be returned with the [`StoreError::InvalidMagicBytes`] error variant in
+    /// its error field and changeset field set to [`Option::None`]
     ///
-    /// [`create_new`]: Store::create_new
-    pub fn open<P>(magic: &[u8], file_path: P) -> Result<Self, FileError>
+    /// If there exist changesets in the file, [`load`] will try to aggregate them in
+    /// a single changeset to verify their integrity. If aggregation fails
+    /// [`StoreErrorWithDump`] will be returned with the [`StoreError::Bincode`] error variant in
+    /// its error field and the aggregated changeset so far in the changeset field.
+    ///
+    /// [`create`]: Store::create
+    /// [`load`]: Store::load
+    pub fn load<P>(magic: &[u8], file_path: P) -> Result<(Self, Option<C>), StoreErrorWithDump<C>>
```

Comment:
```suggestion
    pub fn load<P>(magic: &[u8], file_path: P) -> (Option<Self>, Result<Option<C>, StoreErrorWithDump<C>>)
```

I think we would still like to have access to `Store` even if the last changeset is corrupted in the file.

The only case where we would not like access to `Store` is when the magic bytes do not match (hence the `Option<Self>`).

I'm open for ideas for making this return type more intuitive.

Reply:
I'm dismissing this change in favor of explaining how to recover the `Store` after a failed read.